routes/web/project/location/postgres.rb

```diff
@@ -97,6 +97,13 @@ class CloverWeb
 
         r.redirect "#{@project.path}#{pg.path}"
       end
+
+      r.post "restart" do
+        Authorization.authorize(@current_user.id, "Postgres:create", @project.id)
+        Authorization.authorize(@current_user.id, "Postgres:edit", pg.id)
+        pg.servers.each(&:incr_restart)
```

Comment:
Should we directly restart the PostgresResource servers at the controller, or should we let the resource nexus manage it? If we choose the latter, we can also update its display status to `restarting`.

Reply:
I didn't want to add a new state to the PostgresResource just to increment semaphore of servers there. In the way, still nexus, but the nexus of PostgresServer handles the restart.

Even if we have a separate label in PostgresResource, it would still be problematic to show display status as "restarting" because one of the servers might be busy doing something else.

Reply:
I think this crashes if a server is concurrently removed.

I suppose a case this could in principle be a problem is if the customer needs very precise sequencing and composition of restart with other stuff they are doing, like messing around with GUCs. This is pretty marginal for now though, of complications we wish to expend, it doesn't seem super high on my list.

Reply:
Added a rescue block around incr_restart.